flink-ml-lib/src/main/java/org/apache/flink/ml/classification/knn/KnnParams.java

```diff
@@ -0,0 +1,38 @@
+package org.apache.flink.ml.classification.knn;
+
+import org.apache.flink.ml.param.IntParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.WithParams;
+import org.apache.flink.ml.params.shared.colname.HasFeatureColsDefaultAsNull;
+import org.apache.flink.ml.params.shared.colname.HasKnnDistanceType;
+import org.apache.flink.ml.params.shared.colname.HasLabelCol;
+import org.apache.flink.ml.params.shared.colname.HasPredictionCol;
+import org.apache.flink.ml.params.shared.colname.HasPredictionDetailCol;
+import org.apache.flink.ml.params.shared.colname.HasReservedColsDefaultAsNull;
+import org.apache.flink.ml.params.shared.colname.HasVectorColDefaultAsNull;
+
+/** knn fit parameters. */
+public interface KnnParams<T>
```

Comment:
Does `Knn` and `KnnModel` share exactly the same set of parameters? If not, we have separate parameter classes for them, e.g. `KnnParams` and `KnnModelParams`?

Reply:
use one params is OK. I will refine code later.

Reply:
done